bolt-delegations-cli/src/types.rs

```diff
@@ -0,0 +1,52 @@
+use alloy::signers::k256::sha2::{Digest, Sha256};
+use ethereum_consensus::crypto::{PublicKey as BlsPublicKey, Signature as BlsSignature};
+use serde::Serialize;
+
+#[derive(Debug, thiserror::Error)]
+pub enum KeystoreError {
+    #[error("Failed to read keystore from JSON file {0}: {1}")]
+    ReadFromJSON(String, String),
+    #[error("Failed to decrypt keypair from JSON file {0} with the provided password: {1}")]
+    KeypairDecryption(String, String),
+    #[error("Failed to get public key from keypair: {0}")]
+    UnknownPublicKey(String),
+}
+
+/// Event types that can be emitted by the validator pubkey to
+/// signal some action on the Bolt protocol.
+#[derive(Debug, Clone, Copy)]
+#[repr(u8)]
+enum SignedMessageAction {
+    /// Signal delegation of a validator pubkey to a delegatee pubkey.
+    Delegation,
+}
```

Comment:
Can you add `Revocation` as well? Ideally this CLI should be used for those as well.